### ISSUE TITLE
when upgrading, nextcloud windows client crashes file explorer to avoid reboot, then asks for reboot anyways

### ISSUE THREAD

**Issue by ThatCoffeeGuy** [nextcloud/desktop#4305]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
It should be able to upgrade without wanting to reboot if I select the option that states the same. Or **at least** after crashing explorer.exe, restart it. 

### Actual behaviour
It closes nextcloud, crashes file explorer and then demands a reboot anyways, without bringing file explorer or the taskbar back. I have to manually relaunch explorer.exe from task manager.

### Steps to reproduce
1. Download the newest available Nextcloud client, while there's another one already configured
2. Install, go for "Close the applications and attempt to restart them" 
3. Witness explorer dying and still being asked to reboot your system

### Client configuration
Client version:
Nextcloud-3.4.2-x64

Operating system: Windows 10

OS language: US ENG

Qt version used by client package (Linux only, see also Settings dialog):

Client package (From Nextcloud or distro) (Linux only):

Installation path of client:

C:\Program Files\Nextcloud

### Server configuration

Nextcloud version:

Storage backend (external storage):

### Logs

I have no idea how to generate logs for the installer. 
Please use Gist (https://gist.github.com/) or a similar code paster for longer
logs.

1. Client logfile: 
Since 3.1: Under the "General" settings, you can click on "Create Debug Archive ..." to pick the location of where the desktop client will export the logs and the database to a zip file.
On previous releases: Via the command line: `nextcloud --logdebug --logwindow` or `nextcloud --logdebug --logfile log.txt`
(See also https://docs.nextcloud.com/desktop/3.0/troubleshooting.html#log-files)

2. Web server error log:

3. Server logfile: nextcloud log (data/nextcloud.log):

Pictures:

<img width="386" alt="nice" src="https://user-images.githubusercontent.com/24213618/154820953-a69884c6-cee4-45df-8aab-7c9a721f5ebf.png">

![image](https://user-images.githubusercontent.com/24213618/154820962-d2c40302-a325-445e-8f7c-4d0f3db59514.png)

**Comment by allexzander** [comment]
We've got a corresponding ticket already https://github.com/nextcloud/desktop/issues/3551. It's not that easy. The issue is not happening on Windows 11, only on Windows 10.

**Comment by ThatCoffeeGuy** [comment]
The fix still isn't implemented.

**Comment by Pwuts** [comment]
> We've got a corresponding ticket already #3551. It's not that easy. The issue is not happening on Windows 11, only on Windows 10.

This is not true, I got this behavior on Windows 11 today with the 3.4.4 installer.

**Comment by eibex** [comment]
still relevant

**Comment by ThatCoffeeGuy** [comment]
Okay, today it installed a new version - this time it rebooted my machine without even asking for permission. It said that it "might be needed to reboot" or something like this, but never explicitly said it is going to reboot with a yer or no option (like in the past)

At this point this isn't even funny. I am trying to find alternatives to replace nextcloud.

**Comment by cs-tum** [comment]
I can confirm the unsolicited reboot without any prompt while updating to Desktop Client Version 3.5.2 on Windows 10.0.19044.1766.

**Comment by eibex** [comment]
This is already tracked at #3551

**Comment by claucambra** [comment]
This issue has now been fixed in #4566 . Updates for new versions of the client post-3.5.2 should no longer force reboots. I will now close the issue. Thanks all for reporting!

**Comment by ristein** [comment]
I recently updated to 3.5.4 and observed the exact behavior OP mentioned.
1. Nextcloud Client asks for update and mentions restart necessary
2. Taskbar vanishes and Background gets black
3. Windows is rebooting

**Comment by claucambra** [comment]
> I recently updated to 3.5.4 and observed the exact behavior OP mentioned.
> 
> 1. Nextcloud Client asks for update and mentions restart necessary
> 2. Taskbar vanishes and Background gets black
> 3. Windows is rebooting

3.5.4 was explicitly released to fix this issue -- this means the reboots should stop happening on updates beyond 3.5.4

**Comment by eibex** [comment]
Did you upgrade from 3.6.0 or an older version? 3.6.0 should be fixed.

It asked me to restart but I just pressed No and explorer wasn't crashed (and NC worked just fine even without restarting).
